### DISCUSSION TITLE
search_domain_filter not working, getting citations outside my domain list

### DISCUSSION THREAD

**Discussion by infiloop2** [perplexityai/api-discussion#73] [Q&A]
Hi all, I'm trying to make a api request like this
```
    url = "https://api.perplexity.ai/chat/completions"

    payload = {
        "model": "llama-3.1-sonar-large-128k-online",
        "messages": [
            {
                "role": "system",
                "content": (
                    "Help find the latest news for the user through as many sources as possible. Compile "
                    "it in an unbiased way"
                ),
            },
            {"role": "user", "content": f"What's the latest news about US stock market"},
        ],
        "return_citations": True,
        "search_domain_filter": ["cnbc.com"],
        "search_recency_filter": "day",
    }
```
However I end up getting citations outside cnbc.com domain

**Comment by shubhang98** [comment]
I see. thanks for flagging we will take a look at it.

**Reply by IKDH** [reply]
@shubhang98 have you been able to look at it? It's not working for me either.

**Comment by wyrin-wrynn** [comment]
I am getting same issue: query had these domains:     "search_domain_filter":[
        "ndtv.com",
        "indiatoday.in",
        "thehindu.com",
        "firstpost.com",
        "theprint.in",
        "deccanchronicle.com",
        "financialexpress.com",
        "thehindubusinessline.com",
        "ndtvprofit.com",
        "hindustantimes.com",
        "news18.com",
        "newindianexpress.com"
    ] but the result refernece links were: 

URLs:
(0, 'https://www.indiatoday.in/business/story/stock-market-today-sensex-jumps-1200-points-nifty-3-things-to-know-stock-market-bull-run-2637706-2024-11-22')
(1, 'https://www.business-standard.com/markets/news/sensex-zooms-1200-pts-nifty-atop-23-700-what-s-driving-the-markets-today-124112200596_1.html')
(2, 'https://www.youtube.com/watch?v=4nYRGUmhG5c')
(3, 'https://economictimes.com/markets/stocks/news/bank-stocks-lift-sensex-over-500-points-higher-nifty-back-above-23500/articleshow/115550080.cms')
(4, 'https://economictimes.com/markets/stocks/live-blog/bse-sensex-today-live-nifty-stock-market-updates-22-november-2024/liveblog/115546523.cms')

**Reply by vitalikmay** [reply]
according to [API Reference](https://docs.perplexity.ai/api-reference/chat-completions) `search_domain_filter` is currently limited to only 3 domains for whitelisting and blacklisting.

**Reply by wyrin-wrynn** [reply]
I tried with 3 as well, did not work, even with just 1 whitelisted one it did not work.

**Comment by vitalikmay** [comment]
`search_domain_filter` does not work for domain blacklisting either:
`"search_domain_filter": ["-domain1.com", "-domain2.com", "-domain3.com"]`

**Comment by shubhang98** [comment]
I apologize for the mistake on our documentation but the search_domain_filter is a beta feature. If you don't have access to it the API does not take it into account.

**Reply by vitalikmay** [reply]
is there any chance to get access to the beta? I filled out the form but never heard back

**Reply by IKDH** [reply]
I have also applied few months ago with no answer.

**Reply by IKDH** [reply]
@shubhang98 is it possible to send you an email with our application? Thanks.
